src/renderers/shared/fiber/ReactFiberDevToolsHook.js

```diff
@@ -12,15 +12,13 @@
 
 'use strict';
 
+var warning = require('warning');
+
 import type { Fiber } from 'ReactFiber';
 import type { FiberRoot } from 'ReactFiberRoot';
 
 declare var __REACT_DEVTOOLS_GLOBAL_HOOK__ : Object | void;
 
-if (__DEV__) {
-  var warning = require('warning');
-}
```

Comment:
slight alternative: it looks like 2 of the 3 `warning()` call sites in here are wrapped in `__DEV__`. Would it make sense to wrap the 3rd one in `__DEV__` as well, or unwrap all of them?